### PR TITLE
DOCS-12174 Microsoft Outlook 365 1.0

### PR DESCRIPTION
**Pull request by sathya0**

### CODE REVIEW

**Review by javiermule — DISMISSED** [review]

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-studio.adoc [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+To add an operation for Outlook 365 Connector, follow these steps:
+
+. In the *Mule Palette* view, select *Outlook 365* and then select the desired operation.
+. Drag the operation onto the Studio canvas to the right of the source.
+
+[[configure-global-element]]
+== Configure a Global Element for the Connector
+
+When you configure a connector, configure a global element that all instances of that connector in the app can use. Configuring a global element requires you to provide the authentication credentials that the connector requires to access the target Outlook 365 system. Outlook 365 Connector supports OAuth 2.0 authentication.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-studio.adoc]: I recommend stating the specific types of OAuth authorization here.

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc [hunk 2]
```diff
@@ -0,0 +1,1582 @@
...
+Some operations need prerequisite data, for example, to get a message or
+an event based on its ID, you must create a message or an event first.
+
+[[create-message]]
+== Create Message
+
+This Mule flow creates a message to send later. The message is saved in the
+Drafts folder.
+
+Here is what each operation is doing:
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]:
```suggestion
This example uses the following operations:
```

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc [hunk 3]
```diff
@@ -0,0 +1,1582 @@
...
+. Save the project.
+. From a web browser, test the application by entering `http://localhost:8083/create-msg`.
+
+
+[[delete-message]]
+== Delete Message
+
+This Mule flow deletes an existing message.
+
+Here is what each operation is doing:
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]: Same edit as before - ditto for all relevant places

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc [hunk 4]
```diff
@@ -0,0 +1,1582 @@
...
+* *Transform Message* +
+Transforms the HTTP input used to create a message
+* *Create message* +
+Creates a draft of the message
+* *Transform Message* +
+Outputs the results of the *Create message* operation in JSON format
+* *Logger* +
+Shows the HTTP response from the *Create message* operation
+
+image::create-message.png[Studio Flow for Create Message]
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]: image::create-message.png[Studio flow for creating a message]
or
image::create-message.png[Studio flow for the Get Message operation]

Ditto for all examples

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc [hunk 5]
```diff
@@ -0,0 +1,1582 @@
...
+
+. Verify that your connector is configured.
+. Save the project.
+. From a web browser, test the application by entering `http://localhost:8083/get-msg`.
+
+
+[[list-messages]]
+== List Messages
+
+This Mule flow retrieves all of the messages.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]:
```suggestion
This Mule flow retrieves all of the messages from the signed-in user's mailbox.
```

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc [hunk 6]
```diff
@@ -0,0 +1,1582 @@
...
+
+. Verify that your connector is configured.
+. Save the project.
+. From a web browser, test the application by entering `http://localhost:8083/list-messages`.
+
+
+[[delete-event]]
+== Delete Event
+
+This Mule flow deletes an existing created event.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]:
```suggestion
This Mule flow deletes an existing event.
```

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc [hunk 7]
```diff
@@ -0,0 +1,1582 @@
...
+
+. Verify that your connector is configured.
+. Save the project.
+. From a web browser, test the application by entering `http://localhost:8083/update-event`.
+
+
+[[list-events]]
+== List Events
+
+This Mule flow lists all of the events.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]:
```suggestion
This Mule flow lists all events.
```

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc [hunk 8]
```diff
@@ -0,0 +1,1582 @@
...
+
+. Verify that your connector is configured.
+. Save the project.
+. From a web browser, test the application by entering `http://localhost:8083/create-event`.
+
+
+[[update-event]]
+== Update Event
+
+This Mule flow updates an already existing event.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]:
```suggestion
This Mule flow updates an existing event.
```

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc [hunk 9]
```diff
@@ -0,0 +1,1582 @@
...
+
+. Verify that your connector is configured.
+. Save the project.
+. From a web browser, test the application by entering `http://localhost:8083/get-event-attachment`.
+
+
+[[send-message]]
+== Send Message
+
+This Mule flow sends a message from the Drafts folder. The *Create message*
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]: I don't think you need the second sentence. You could say:

This Mule flow sends an existing message from the Drafts folder.

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc [hunk 10]
```diff
@@ -0,0 +1,1582 @@
...
+
+. Verify that your connector is configured.
+. Save the project.
+. From a web browser, test the application by entering `http://localhost:8083/send-message`.
+
+
+[[send-mail]]
+== Send Mail
+
+This Mule flow sends an email without creating a message first.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]: This is the first time "email" is used in the descriptions. Maybe instead:

This Mule flow sends the message specified in the request body.

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc [hunk 11]
```diff
@@ -0,0 +1,1582 @@
...
+
+. Verify that your connector is configured.
+. Save the project.
+. From a web browser, test the application by entering `http://localhost:8083/create-reply-all-to-message`.
+
+
+[[reply-all-to-message]]
+== Reply All to Message
+
+This Mule flow creates a reply-all message that is sent without creating it first
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]: Not clear - where is the actual message used in the reply all coming from?

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc [hunk 12]
```diff
@@ -0,0 +1,2559 @@
...
+| Listener Config a| String |  Configuration for the HTTP listener that listens for requests on the access token callback endpoint. |  | x
+| Callback Path a| String |  Path of the access token callback endpoint. |  | x
+| Authorize Path a| String |  Path of the local HTTP endpoint that triggers the OAuth dance. |  | x
+| External Callback Url a| String |  URL that the OAuth provider uses to access the callback endpoint if the endpoint is behind a proxy or accessed through an indirect URL. |  |
+| Object Store a| String |  Configures the object store that stores data for each resource owner. If not configured, Mule uses the default object store. |  |
+|===
+[[Outlook365Config_OauthClientCredentials]]
+===== OAuth v2.0 Client Credentials
+
+The OAuth 2.0 client credentials grant accesses web-hosted resources by using the identity of the application.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc:67]: The OAuth 2.0 client credentials grant access to web-hosted resources by using the identity of the application.

Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc]:
```suggestion
The OAuth 2.0 client credentials grant access to web-hosted resources by using the identity of the application.
```

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc [hunk 13]
```diff
@@ -0,0 +1,2559 @@
...
+* OUTLOOK365:TIMEOUT
+* OUTLOOK365:VALIDATION
+
+
+[[GetEvent]]
+== Get Event
+`<outlook365:get-event>`
+
+
+Gets the properties and relationships of the specified event object. Currently, this operation returns event bodies in only HTML.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc]:
```suggestion
Gets the properties and relationships of the specified event object. Currently, this operation returns event bodies in HTML only.
```

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc [hunk 14]
```diff
@@ -0,0 +1,2559 @@
...
+* OUTLOOK365:TIMEOUT
+* OUTLOOK365:VALIDATION
+
+
+[[SendMessage]]
+== Send Message
+`<outlook365:send-message>`
+
+
+Sends a message in the draft folder. The draft message can be a new message draft, reply draft, reply-all draft, or a forward draft. The message is then saved in the Sent Items folder.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc]:
```suggestion
Sends a message in the draft folder. The draft message can be a new message draft, reply draft, reply-all draft, or forward draft. The message is then saved in the Sent Items folder.
```

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc [hunk 15]
```diff
@@ -0,0 +1,2559 @@
...
+| Reconnection a| <<Reconnection>> |  Configures a reconnection strategy to use when a connector operation fails to connect to an external server. |  |
+| Consumer Key a| String | OAuth consumer key, as registered with the service provider. |  | x
+| Consumer Secret a| String |  OAuth consumer secret, as registered with the service provider. |  | x
+| Authorization Url a| String |  URL of the service provider's authorization endpoint. |  https://login.microsoftonline.com/{tenant}/oauth2/v2.0/authorize |
+| Access Token Url a| String |  URL of the service provider's access token endpoint. |  https://login.microsoftonline.com/{tenant}/oauth2/v2.0/token |
+| Scopes a| String |  OAuth scopes to request during the OAuth dance. This value defaults to the scopes in the annotation. |  |
+| Resource Owner Id a| String |  Resource owner ID to use with the authorization code grant type. |  |
+| Before a| String |  Name of the flow to execute immediately before starting the OAuth dance. |  |
+| After a| String |  Name of the flow to execute immediately after receiving an accessToken. |  |
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc]:
```suggestion
| After a| String |  Name of the flow to execute immediately after receiving an access token. |  |
```

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc [hunk 16]
```diff
@@ -0,0 +1,2559 @@
...
+* OUTLOOK365:TIMEOUT
+* OUTLOOK365:VALIDATION
+
+
+[[GetMessageAttachment]]
+== Get Message Attachment
+`<outlook365:get-message-attachment>`
+
+
+Reads the properties and relationships of an attachment that is attached to a user message. You can append the path segment /$value to get the raw contents of a file or item attachment.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc]: Use backticks for /$?

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc [hunk 17]
```diff
@@ -0,0 +1,2559 @@
...
+| Top a| Number |  |  |
+|===
+
+[[RepeatableInMemoryIterable]]
+=== Repeatable In Memory Iterable
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Initial Buffer Size a| Number | The number of instances to initially keep in memory to consume the stream and provide random access to it. If the stream contains more data than can fit into this buffer, then the buffer expands according to the bufferSizeIncrement attribute, with an upper limit of maxInMemorySize. | 100 |
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc]:
```suggestion
| Initial Buffer Size a| Number | The number of instances to initially keep in memory to consume the stream and provide random access to it. If the stream contains more data than can fit into this buffer, then the buffer expands according to the *Buffer Size Increment* field, with an upper limit of maxInMemorySize. | 100 |
```

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc [hunk 18]
```diff
@@ -0,0 +1,2559 @@
...
+* OUTLOOK365:NOT_FOUND
+* OUTLOOK365:RETRY_EXHAUSTED
+* OUTLOOK365:SERVER_ERROR
+* OUTLOOK365:TIMEOUT
+* OUTLOOK365:VALIDATION
+
+
+== Sources
+
+[[ModifiedEventListener]]
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc:2100]: If this is NOT a REST connector, all of the sources should have a description.

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc (hunk 9 above)
Comment (sathya0) [reply microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]: This may be important actually, there was a note on line 38-40 about how some operations need prerequisite operations in order to use them. I tried rewording it so let me know how it looks now

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc (hunk 10 above)
Comment (sathya0) [reply microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]: email/message are two different things so I guess that is why they clarified the email was sent without creating a message

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc (hunk 11 above)
Comment (sathya0) [reply microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]: clarified

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc (hunk 12 above)
Comment (sathya0) [reply microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc]: looks like the name itself is "OAuth 2.0 Client Credentials Grant" so grant is part of the name instead of a verb

Comment (dhirsch1) [reply microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc]: How about this wording to avoid the confusion?

The OAuth 2.0 client credentials grant gets access to...

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc [hunk 19]
```diff
@@ -2101,6 +2101,7 @@ Updates the properties of a message object.
 == On Modified Event
 `<outlook365:modified-event-listener>`
 
+Triggers a flow for modified events.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc]: I think this wording is better. Regarding trigger vs initiate - they can be interchangeable but because some of the docs call sources trigger, I avoid the term in descriptions - up to you whether to use it or not.

Initiates a flow when an event is modified.

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc [hunk 20]
```diff
@@ -2151,6 +2152,7 @@ Updates the properties of a message object.
 == On New Email
 `<outlook365:new-email-listener>`
 
+Triggers a flow for new received emails.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc]: Initiates a flow when a new email is received.

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc [hunk 21]
```diff
@@ -2200,6 +2202,7 @@ Updates the properties of a message object.
 == On New Event
 `<outlook365:new-event-listener>`
 
+Triggers a flow for new events.
```
Comment (dhirsch1) [inline microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc]: Initiates a flow when a new event is detected.

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc (hunk 10 above)
Comment (dhirsch1) [reply microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]: OK.  Feel free to ignore my musings:

We still mention the message in the request body, so this wording makes it sound like message has two different meanings in this doc (one in an email and one in a request body). I'm guessing the Create Message operation creates a message in an email. If not, then your wording is fine. If so, then the saying "This Mule flow sends an email without creating a message first." doesn't really make sense.

**Review by dhirsch1 — DISMISSED** [review]

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc (hunk 19 above)
Comment (sathya0) [reply microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-reference.adoc]: I like yours a lot better!

File: microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc (hunk 10 above)
Comment (sathya0) [reply microsoft-outlook-365/1.0/modules/ROOT/pages/microsoft-outlook-365-connector-examples.adoc]: Yeah, Create Message is actually for creating a draft message in a request body, not the message in the email. Basically this flow sends an email without creating a draft first

**Review by dhirsch1 — APPROVED** [review]
